v0/util/cache.js

```diff
@@ -17,12 +17,20 @@ class Cache {
 
     return storeFunction().then(result => {
       // store in cache if the value is valid, else skip
+      if (result && typeof result === "object") {
+        this.cache.set(key, result.token, result.age);
+        return result.token;
+      }
```

Comment:
This logic should not be specific to tokens.

Reply:
Yes, it can be used for any data not specifically tokens. Will update the namings to avoid confusion.